src/scripts/hooks/use-mouse-move.ts

```diff
@@ -0,0 +1,15 @@
+import {useEffect, useState} from 'react';
+
+export const useMouseMove = () => {
+  const [mouseMove, setMouseMove] = useState(true);
+
+  useEffect(() => {
+    let timer = 0;
+    window.onmousemove = () => {
```

Comment:
use window.addEventListener instead of onmousemove and removeEventListener in the clean up function. Using onmousemove function can cause errors if anything else relies on this function. addEventlistener is the more general solution

Reply:
Also the cleanup method should clear the timeout too